git-extra/git-update

```diff
@@ -5,6 +5,13 @@
 # when confirmation to do so is given.
 
 git_update () {
+	proxy=$(proxy-lookup https://api.github.com | grep URL | awk '{ print $4}' | awk -F";" '{print $1}')
+	if test "$proxy" != ""
+	then
+		export https_proxy="http://$proxy"
+		echo "Using proxy server $https_proxy detected from Internet Explorer settings"
+	fi
+
```

Comment:
Would it work if you set `CURL_SSL_BACKEND=schannel` specifically for the `curl.exe` invocation and *not* try to discover the proxy ourselves?

Reply:
I've tested that configuration. libcurl does not automatically detect proxy when using schannel backend. see also 
https://curl.haxx.se/docs/faq.html#Does_curl_support_Javascript_or

Reply:
That is funny, because all the claims I read on the internets (e.g. https://stackoverflow.com/questions/1600743/could-not-create-ssl-tls-secure-channel-could-the-problem-be-a-proxy-server) seem to suggest that Secure Channel respects the Internet Options, including the proxy setting...

Reply:
As far as I understood until now, schannel does only SSL/TLS, see https://msdn.microsoft.com/en-us/library/windows/desktop/aa380123(v=vs.85).aspx

Reply:
Ah, you're right, of course, it is cURL's duty to handle the higher-order aspects of the HTTPS protocol (which a proxy kinda is).

I wonder, however, if we can do better than this and integrate that proxy lookup into cURL. And indeed, we talked about this before: https://github.com/git-for-windows/git/issues/387

And of course you know this because you referenced that very issue in the header of the file...

Now, I wonder further how any of your `git fetch` calls work if you have a proxy and not set the  environment variable `https_proxy` globally... And my guess is that you use the Git config variable `http.proxy` to this end.

So naturally, I wonder, whether we should not use that `http.proxy` config setting here, too, instead of adding a helper to look up the system proxy setting *just* for `git update` and for nothing else.

Reply:
Yes, having proxy detect directly within curl and git would be my favorite => no http.proxy setting within git and no proxy variable for curl.

Using `http.proxy` could be an option, however updater will fail if it is not configured...

Reply:
I feel we should go for the `http.proxy`-only way (being careful *not* to set `https_proxy` if it has been already set, or if `http.proxy` is not configured or has an empty value).

We can then revive the effort to get automatic proxy.detection into cURL.

Reply:
(part of my rationale: if `http.proxy` is not configured but required, cloning/fetching via HTTPS will already fail)

Reply:
@dscho I've made https://github.com/git-for-windows/build-extra/pull/155 which uses `http.proxy` and no proxy-lookup magic.

Reply:
For the record: if you force-push your branch, that updates the original Pull Request. No need to add new ones...